src/conversions/length_conversion.rs

```diff
@@ -0,0 +1,139 @@
+/// by first converting it to meter and than convert it to destination unit
+pub fn length_conversion(input: f64, from: LengthUnit, to: LengthUnit) -> f64 {
+    input * from.get_unit_to_meter_multiplier() * to.get_unit_from_meter_multiplier()
+}
+
+#[cfg(test)]
+mod length_conversion_tests {
+    use std::collections::HashMap;
+
+    use super::LengthUnit::*;
+    use super::*;
+
+    #[test]
+    fn meter_to_other() {
+        assert_eq!(length_conversion(4f64, Meter, Millimeter), 4000.0);
+        assert_eq!(length_conversion(4f64, Meter, Foot), 13.123359580052492);
+        assert_eq!(length_conversion(1.0, Meter, Kilometer), 0.001);
+    }
+
+    #[test]
+    fn other_to_meter() {
+        assert_eq!(length_conversion(4f64, Millimeter, Meter), 0.004);
+        assert_eq!(length_conversion(2.0, Foot, Meter), 0.6096);
+        assert_eq!(length_conversion(1.0, Inch, Meter), 0.0254);
+        assert_eq!(length_conversion(4.0, Yard, Meter), 3.6576);
+        assert_eq!(length_conversion(3.0, Foot, Meter), 0.9144000000000001);
+    }
+
+    #[test]
+    fn other_to_other() {
+        assert_eq!(length_conversion(1.0, Kilometer, Inch), 39370.07874015748);
+        assert_eq!(length_conversion(3.0, Kilometer, Mile), 1.8641182099494205);
+        assert_eq!(length_conversion(4.0, Foot, Yard), 1.3333333333333335);
+        assert_eq!(length_conversion(2.0, Inch, Mile), 3.156573502181019e-5);
+        assert_eq!(length_conversion(2.0, Centimeter, Millimeter), 20.0);
+        assert_eq!(
+            length_conversion(2.0, Centimeter, Yard),
+            0.021872265966754158
+        );
+        assert_eq!(length_conversion(4.0, Yard, Kilometer), 0.0036576);
+        assert_eq!(length_conversion(3.0, Foot, Inch), 36.00000000000001);
+        assert_eq!(length_conversion(4.0, Mile, Kilometer), 6.43736);
+        assert_eq!(length_conversion(2.0, Mile, Inch), 126719.68503937007);
+        assert_eq!(length_conversion(3.0, Millimeter, Centimeter), 0.3);
+        assert_eq!(
+            length_conversion(3.0, Millimeter, Inch),
+            0.11811023622047245
+        );
+    }
+
+    #[test]
+    fn zero_to_zero() {
+        let units = vec![
+            Millimeter, Centimeter, Meter, Kilometer, Inch, Foot, Yard, Mile,
+        ];
+
+        for u1 in units.clone() {
+            for u2 in units.clone() {
+                assert_eq!(length_conversion(0f64, u1, u2), 0f64);
+            }
+        }
+    }
+
+    #[test]
+    fn length_of_one_meter() {
+        let map = HashMap::from([
+            (Millimeter, 1000f64),
+            (Centimeter, 100f64),
+            (Kilometer, 0.001f64),
+            (Inch, 39.37007874015748f64),
+            (Foot, 3.280839895013123f64),
+            (Yard, 1.0936132983377078f64),
+            (Mile, 0.0006213727366498068f64),
+        ]);
+
+        for (k, v) in map {
+            assert_eq!(length_conversion(1f64, Meter, k), v);
+        }
+    }
```

Comment:
```suggestion
    #[test]
    fn length_of_one_meter() {
        let meter_in_different_units = HashMap::from([
            (Millimeter, 1000f64),
            (Centimeter, 100f64),
            (Kilometer, 0.001f64),
            (Inch, 39.37007874015748f64),
            (Foot, 3.280839895013123f64),
            (Yard, 1.0936132983377078f64),
            (Mile, 0.0006213727366498068f64),
        ]);

        for (input_unit, input_value) in &meter_in_different_units {
            for (target_unit, target_value) in &meter_in_different_units {
                assert!(
                    num_traits::abs(
                        length_conversion(*input_value, *input_unit, *target_unit) - *target_value
                    ) < 0.0000001
                );
            }
        }
    }
```
Please notice, that the `hash_map` could be a _variable_.

Reply:
Ok!
Its here:
https://github.com/ali77gh/TheAlgorithmsRust/commit/ea60fed02ae820e8968b6c8ce8e7c205cd6f66b1